README.md

```diff
@@ -158,17 +177,37 @@ PowerShell.
 
 # Implementation
 
-PxGet uses its own private copies of the `PackageManagement` and `PowerShellGet` modules, which it uses to find and save modules. It calls `Find-Module` once to get the latest version of all the modules in the "pxget.json" file. For each 
-module not pinned to the latest version, PxGet will call `Find-Module` again to get all versions of that module. (If the
-module's version from the "pxget.json" file contains the prerelease suffix, `-`, prerelease versions will be included.) PxGet selects the first version returned by `Find-Module` that matches the version wildcard from the "pxget.json" file.
-Each module is installed with `Save-Module`, from the repository on the module object returned by `Find-Module`.
+## Overview
+
+PxGet uses its own private copies of the `PackageManagement` and `PowerShellGet` modules to find and install modules.
+
+## pxget lock
+
+It calls
+`Find-Module` once to get the latest version of all the modules in the "pxget.json" file. For each module with a
+specific version that doesn't match the latest version, PxGet will call `Find-Module` again to get all versions of that
+module. (If the module's version from the "pxget.json" file contains the prerelease suffix, `-`, prerelease versions
+will be included.) PxGet selects the first version returned by `Find-Module` that matches the version wildcard from the
+"pxget.json" file. It writes these versions to a lock file, which is used by the `pxget install` command to install
+the modules.
+
+## pxget install
+
+For each module and version in the lock file, `pxget install` calls the `Save-Module` cmdlet to install that specific
+version. It passes the name of all installed repositories to the `Save-Module` cmdlet's `-Repository` parameter. The
+`Save-Module` command loops through each repository, and installs the first module it finds.

```

Comment:
I think it's important that there be a way to specify a registry to use for certain or all modules, otherwise simply installing the first module found in multiple repositories makes one vulnerable to [Dependency Confusion](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610).